core/src/mindustry/ui/dialogs/JoinDialog.java

```diff
@@ -445,13 +450,29 @@ public void connect(String ip, int port){
             logic.reset();
             net.reset();
             Vars.netClient.beginConnecting();
-            net.connect(ip, port, () -> {
+            net.connect(lastIp = ip, lastPort = port, () -> {
                 hide();
                 add.hide();
             });
         });
     }
 
+    public void reconnect(){
+        ui.loadfrag.show("@reconnecting");
+
+        this.ping = Timer.schedule(() -> {
```

Comment:
maybe use `ping = ....` here? As `this` is unnecessary

Reply:
no, i tried that, but then java throws several huge tantrums about it not being initialized / not final / several other weird errors / etc, making it a class property was the cleanest working solution